src/samples/tester.cpp

```diff
@@ -0,0 +1,232 @@
+    member_b->Join("Member B");             // Change the ncikname to "Member A" to test name collision
+
+    while ((current_state = member_b->GetState()) != RoomMember::State::Joined) {
+        switch (current_state) {
+        case RoomMember::State::Error:
+            std::cout <<  "Unkown error while connecting Member B"<< std::endl;
+            return 1;
+        case RoomMember::State::LostConnection:
+            std::cout <<  "Lost connection while connecting Member B"<< std::endl;
+            return 1;
+        case RoomMember::State::MacCollision:
+            std::cout <<  "MAC collision of Member B"<< std::endl;
+            return 1;
+        case RoomMember::State::NameCollision:
+            std::cout <<  "Name collision of Member B"<< std::endl;
+            return 1;
+        case RoomMember::State::RoomFull:
+            std::cout <<  "Room is to crowded"<< std::endl;
+            return 1;
+        case RoomMember::State::RoomDestroyed:
+            std::cout <<  "Room is dead"<< std::endl;
+            return 1;
+        case RoomMember::State::WrongVersion:
+            std::cout <<  "You got the wrong version"<< std::endl;
+            return 1;
+        case RoomMember::State::Idle:
+            std::cout <<  "This should not happen!"<< std::endl;
+            return 1;
+        default:
+            sleep(1);
+            break;
+        }
+    }
+    std::cout << "Successfully joined Member B" << std::endl;
+
+    member_list = member_a->GetMemberInformation();
+    for (const auto member:member_list) {
+        std::ostringstream mac_address;
+        for (const uint8_t& value: member.mac_address) {
+            mac_address << std::hex << std::setw(2) << std::setfill('0') << +value << ":";
+        }
+        std::cout << member.nickname << " " << mac_address.str() << " " << member.game_name << std::endl;
+    }
+
+    RoomInformation room_information = member_a->GetRoomInformation();
+    std::cout << room_information.name << "(" << member_list.size() << "/" << room_information.member_slots << ")" << std::endl;
+
+/*   Chat isn't in master yet
+    // Test: Chat
+    std::cout << std::endl;
+    std::cout << "Testing the chat" << std::endl << std::endl;
+
+    member_a->SendChatMessage("This is a test message");
+    std::deque<RoomMember::ChatEntry> chat_entries;
+
+    std::cout << "This is what Member B receives:" << std::endl;
+    while(chat_entries.size() == 0) {
+        chat_entries = member_b->PopChatEntries();
+        sleep(10);
+    }
+    for (const auto& entry: chat_entries) {
+    std::cout << entry.nickname << ": " << entry.message << std::endl; ;
+    }
+
+    std::cout << "Member A should receive it, too:" << std::endl;
+    while(chat_entries.size() == 0) {
+        chat_entries = member_a->PopChatEntries();
+        sleep(10);
+    }
+    for (const auto& entry: chat_entries) {
+    std::cout << entry.nickname << ": " << entry.message << std::endl;
+    }
+*/
```

Comment:
Instead of having dead code just base your work on the commits you need and mention it in the PR description.
Even better: split it in a seperate commit and keep that one local for now